awsapilib/controltower/controltower.py

```diff
@@ -524,18 +581,75 @@ def delete_organizational_unit(self, name: str) -> bool:
         self.logger.debug(response)
         return bool(response.get('ResponseMetadata', {}).get('HTTPStatusCode') == 200)
 
+    @staticmethod
+    def _validate_hierarchy(hierarchy):
+        if not isinstance(hierarchy, (list, tuple)):
+            raise InvalidParentHierarchy(f'Only "list" is a valid argument, received "{type(hierarchy)}"')
+        if isinstance(hierarchy, tuple):
+            hierarchy = list(hierarchy)
+        if not hierarchy[0] == 'Root':
+            hierarchy.insert(0, 'Root')
+        hierarchy = [ou for ou in hierarchy if ou]
+        if len(hierarchy) > OU_HIERARCHY_DEPTH_SUPPORTED:
+            raise InvalidParentHierarchy(f'Only {OU_HIERARCHY_DEPTH_SUPPORTED} levels are supported under Root OU, '
+                                         f'received {hierarchy} that is {len(hierarchy)}')
+        return hierarchy
+
+    @staticmethod
+    def _get_ou_by_attribute_pairs(ou_container, attribute_pairs):
+        return next((ou for ou in ou_container
+                     if all([getattr(ou, name) == value for name, value in attribute_pairs.items()])), None)
+
+    def _get_final_parent_ou(self, organizational_units, hierarchy, parent_ou_id=None, force_create=False):
+        organizational_unit = None
+        parent_ou_name = None
+        working_path = []
+        for name in hierarchy:
+            attributes = {'name': name,
+                          'parent_ou_id': parent_ou_id}
+            organizational_unit = ControlTower._get_ou_by_attribute_pairs(organizational_units, attributes)
+            if all([not organizational_unit, not force_create]):
+                raise NonExistentOU(f'No OU with name "{name}" and parent OU name "{parent_ou_name}" '
```

Comment:
parent_ou_name will be always None here. parent_ou_id should be sufficient?

Reply:
parent_ou_name will only be None if the parent OU is Root, In all other cases it should be properly populated.